easybuild/easyblocks/s/siesta.py

```diff
@@ -108,6 +108,12 @@ def configure_step(self):
             (r"^(FPPFLAGS_free_F90\s*=.*)$", r"\1 -ffree-line-length-none"),
         ]
 
+        gfortran_flags = ''
+        gcc_version = get_software_version('GCCcore') or get_software_version('GCC')
+        if LooseVersion(gcc_version) >= LooseVersion('10.0'):
+            # -fallow-argument-mismatch is required when compiling with GCC 10.x & more recent
+            gfortran_flags = '-fallow-argument-mismatch'
```

Comment:
Hmm, not sure this is a good approach...

Surely Siesta will be updated such that `-fallow-argument-mismatch` is no longer needed, so this really belongs in the easyconfig no?

Reply:
Currently the Siesta EasyBlock does not allow to set this parameter in the EasyConfig (at least I did not manage). That is why i have added it to the EasyBlock.

A similar approach is used in
- BerkeleyGW https://github.com/easybuilders/easybuild-easyblocks/blob/develop/easybuild/easyblocks/b/berkeleygw.py#L122
- CP2K https://github.com/easybuilders/easybuild-easyblocks/blob/develop/easybuild/easyblocks/c/cp2k.py#L551-L559
- QuantumESPRESSO https://github.com/easybuilders/easybuild-easyblocks/blob/develop/easybuild/easyblocks/q/quantumespresso.py#L184-L185

Reply:
Ah, ok, forgot about that... In that case, my complaint is moot :)

Reply:
I agree to set this in the easyblock but this needs to be constrained not only by version of compiler, but also by version of SIESTA. If the fututure version with the fix is still not known, we should set an upper limit to the current major version which will force us to not forget about this setting and review it in the future.

Reply:
@SebastianAchilles Let's include @lexming's suggestion?

Reply:
@SebastianAchilles maybe also report this upstream?

Reply:
@SebastianAchilles ping regarding including lexming's suggestions

Reply:
fixed in ec82d64